### PR TITLE
SE-6932 SAP SNC Connection

### PR DESCRIPTION
**Pull request by dhirsch1**

### CODE REVIEW

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+. Select the *SNC SAPCryptolib* folder in the left pane.
++
+The SAP distinguished name and certificate list are displayed.
+
+=== Link the SAP Certificate to Your Certificate
+
+Download the SAP instance certificate and link it to your certificate:
+
+. Click the glasses icon (first icon on the left of the screen) to enable the edition mode, and then double click `CN=SL-ABAP-IDP`.
+. Click the download button at the bottom of the screen and enter the target path for the download. *(Reviewers: is this text correct? Also, is the download a button or icon?)*
```
Comment (VaneMolina) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]: It is a button, which is enable when you click on the glasses icon (edition mode)

**Comment by sorinmare** [comment]
Hi @dhirsch1 ,
I discussed with @VaneMolina and apart the above comment the documentation it is fine

**Review by sorinmare — DISMISSED** [review]
LGTM

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 2]
```diff
@@ -0,0 +1,170 @@
+= Configure an SAP Secure Network Communication Connection
+
+Configure an SAP Secure Network Communication (SNC) to take advantage of security features such as secure data communication between the SAP system client and server, application level end-to-end security, and the ability to change security products without impacting your SAP business applications. Enabling SNC requires you to configure the client environment, the SAP server, and Anypoint Connector for SAP (SAP Connector).
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
Configure an SAP Secure Network Communication (SNC) to take advantage of security features such as secure data communication between the SAP system client and server, application-level end-to-end security, and the ability to change security products without impacting your SAP business applications. Enabling SNC requires you to configure the client environment, the SAP server, and Anypoint Connector for SAP (SAP Connector).
```

Comment (kbostrom) [reply sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]: LOVE the last sentence, which prepares audiences for the three subtopics.

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+Configure the system variables that depend on the operating system you are using, and generate the personal security environment (PSE):
+
+. Open the Windows command line or Mac Terminal app.
+. Use the `SAPCAR` utility to unpack the SAP Cryptographic Library to the specified folder:
++
+`SAPCAR –xvf <folder-name.SAR>`
++
+. Configure the following environment variables:
+** SNC_LIB  `<path_to_library>\sapcrypto.dll`
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
** SNC_LIB: `<path_to_library>\sapcrypto.dll`
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+** SECUDIR = `<path_to_library>`
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
** SECUDIR: `<path_to_library>`
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+** LD_LIBRARY_PATH= `<path_to_library>\sapcrypto.dll`
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
** LD_LIBRARY_PATH: `<path_to_library>\sapcrypto.dll`
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+** SSF_LIBRARY_PATH= `<path_to_library>\sapcrypto.dll`
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
** SSF_LIBRARY_PATH: `<path_to_library>\sapcrypto.dll`
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+`CN=RFC, C=AR, O=YourCompany, OU=IT`
++
+You can use STRUST to see names that are already in use.
++
+. Use the `sapgenpse` tool to create the `cred_v2` file, which stores the credentials for accessing the PSE:
++
+`sapgenpse seclogin -p RFC -O <user-credentials>`
+
++
+. Use the `sapgenpse` tool to export the client’s certificate from the PSE.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
. Use the `sapgenpse` tool to export the client’s certificate from the PSE:
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
++
+`sapgenpse <export_own_cert> –v –p RFC –o RFC_MULE.crt`
+
+== Configure the SAP Server
+
+Configure the certificates and tables that authorize you to use SNC with your distinguished name. To configure the SAP server:
+
+. View the SAP distinguished name and certificates.
+. Link the SAP certificate to your certificate.
+. Authorize the SNC client in SAP.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:60]: Yaaaaas! Sign-posting!

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+
+=== View the SAP Distinguished Name and Certificates
+
+View the SAP distinguished name and list of certificates:
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:64]: Any way to make this just a bit different from the title?

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+To configure an SAP SNC, you must have access to:
+
+* The SAP GUI application, which you can download from the https://sites.google.com/a/mulesoft.com/sap/abc/libs-tools?authuser=0[MuleSoft SAP Resource Center]
+* The SAPCAR utility and SAP Cryptographic Library, which you can download from the SAP Support Portal, if you are an S-user
+* An SAP ECC instance that can use SNC system variables
+* The following transactions:
+** `STRUST` (Trust Manager)
+** `SM30` (Table Maintenance)
+
+== Configure the Client Environment
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:16]: At some point, we should start using the Title+Intro, Prereqs+intro+list, and Task+intro+Steps formats....

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+
+View the SAP distinguished name and list of certificates:
+
+. Log in to the SAP GUI and enter the `strust` transaction.
+. Select the *SNC SAPCryptolib* folder in the left pane.
++
+The SAP distinguished name and certificate list are displayed.
+
+=== Link the SAP Certificate to Your Certificate
+
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:72]: GLOBALLY consider implementing Title+Intro+Steps

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+Download the SAP instance certificate and link it to your certificate:
+
+. Click the glasses icon (first icon on the left of the screen) to enable the edition mode, and then double click `CN=SL-ABAP-IDP`.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:75]: GLOBALLY consider replacing descriptions with actual icon screen caps: "the glasses icon (first icon on the left of the screen" > image

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+. Click the download button at the bottom of the screen and enter the target path for the download.
+. Click the import icon and enter the target path to upload your certificate.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:77]: SHOW the icon?

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+. Click *Add to Certificate List* and then click the save icon.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:78]: SHOW the icon?

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+. Open a cmd or terminal window and use the `sapgenpse` tool to link your certificate to your PSE:
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
. Open a command-line interface or terminal window and use the `sapgenpse` tool to link your certificate to your PSE:
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+. In the SAP GUI, navigate to the *SM30* transaction and enter `VSNCSYSACL` as the view name.
+. Click *Maintain*.
+. Select *E* as the work area.
+. In the *System ID* field, enter the SAP ID.
+. In the *SNC name* field, enter your distinguished name:
++
+`p:<your-distinguished-name>`
++
+. Select all of the checkboxes to activate the entries.
+. Click the save icon.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:100]: SHOW the icon?

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
++
+* If the server does not accept the SNC name, check that the name used on the PSE file is the same as the name used on the Mule app and the name listed on STRUST.
++
+. Enable the SAP user for the Mule app so the app can use the SNC connection type:
+	.. Navigate to the *SM30* transaction, specify the `VUSREXTID` view, and click *Maintain*.
+	.. In the *External ID type* field, select `DN` as the work area.
+	.. In the *External ID* field, enter the distinguished name.
+	.. In the *User* field, enter the SAP username.
+	.. Select *Activated*.
+	.. Click the save icon.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:112]: ditto; global re. icon types

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 18]
```diff
@@ -0,0 +1,170 @@
...
+* *SAP system number*
+* *SAP client ID*
+* *Application server host*
+.. On the *Advanced* tab, configure extended properties.
+
++
+For an X.509 certificate connection, follow these steps:
+. In the *Connection* field, select `Certificate`.
+. On the *General* tab, configure these fields:
+** *X.509 Certificate* (not base64 encoded)
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:141]: "Base64"?

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 19]
```diff
@@ -0,0 +1,170 @@
...
+** *Sap client ID*
+** *Application server host*.
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
** *Application server host*
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 20]
```diff
@@ -0,0 +1,170 @@
...
+.. On the *Advanced* tab, configure the extended properties.
+
+=== SAP Server Side Extended Properties Example:
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
=== SAP Server-Side Extended Properties Example:
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 21]
```diff
@@ -0,0 +1,170 @@
...
+
+The following table shows example values for the server side extended properties:
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc]:
```suggestion
The following table shows example values for the server-side extended properties:
```

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 22]
```diff
@@ -0,0 +1,170 @@
...
+
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:149]: We really should not be using tables for item+description or item+value, per the style reference.

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc [hunk 23]
```diff
@@ -0,0 +1,170 @@
...
+|jco.server.snc_partnername| `p:CN=SL-ABAP-IDP`
+|jco.server.snc_qop| `8`
+|jco.server.snc_myname| `p:CN=MVIDP, C=AR, O=MuleSoft, OU=IT`
+|jco.server.snc_lib| `C:/SNC/64bit/sapcrypto.dll`
+|===
+
+=== SAP Client Side Extended Properties Example
+
+The following table shows example values for the client side extended properties:
+
```
Comment (kbostrom) [inline sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:163]: ditto re. tables

**Review by kbostrom — DISMISSED** [review]
Edited per scope of request

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc (hunk 18 above)
Comment (dhirsch1) [reply sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:141]: I've seen it mostly with lowercase

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc (hunk 22 above)
Comment (dhirsch1) [reply sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:149]: We don't agree. Connector docs use tables and we want to be consistent. This is a bigger discussion.

File: sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc (hunk 11 above)
Comment (dhirsch1) [reply sap/5.1/modules/ROOT/pages/sap-connector-config-snc.adoc:72]: I don't have access to the functionality.

**Review by sorinmare — DISMISSED** [review]
LGTM

**Review by FilipBogdanSV — DISMISSED** [review]
LGTM

**Review by FilipBogdanSV — APPROVED** [review]
Looks good.

**Review by kbostrom — APPROVED** [review]
Reapproval without reediting

**Pull request merged**